### DISCUSSION TITLE
How to restrict dragging nodes by an axis?

### DISCUSSION THREAD

**Discussion by ayZagen** [bcakmakoglu/vue-flow#904] [Q&A] [answered]
I would like to have a node to be dragged only vertically or horizontally. I tried playing with events but it didn't work. In d3 we could override drag behavior inside 'drag' event but it seems it does not work here. here is a node example i tried:

```js
  {
    id: 'id1',
    label: 'drag only vertivally',
    position: { x: 100, y: 100 },
    events: {
      drag({ event }) {
        if (event.movementX !== 0) {
          event.preventDefault() //no effect
        }
      },
    },
    style: { backgroundColor: 'rgba(16, 185, 129, 0.5)', width: '600px', height: '200px' },
  } as Node,
```

**Comment by bcakmakoglu** [comment, marked as answer]
There's no option to restrict axis of movement.
You could do this by handling node changes yourself something like this

```vue
onNodesChange((changes) => {
  const positionChanges = changes.filter((change) => change.type === 'position') as NodePositionChange[]

  if (positionChanges.length) {
    applyNodeChanges(
      positionChanges.map((change) => {
        return {
          ...change,
          position: {
            y: change.from.y,
            x: change.position.x,
          },
        }
      }),
    )
  } else {
    applyNodeChanges(changes)
  }
})
```

```html
<VueFlow v-model="elements" :apply-default="false" />
```

[Here's an example](https://codesandbox.io/p/sandbox/vigorous-rubin-pf3rkn?file=%2Fsrc%2FApp.vue&selection=%5B%7B%22endColumn%22%3A10%2C%22endLineNumber%22%3A20%2C%22startColumn%22%3A10%2C%22startLineNumber%22%3A20%7D%5D)

**Reply by ayZagen** [reply]
Thanks, I got the idea but the `positionChanges` contains incorrect values for nested nodes. `position` contains relative path to parent but `from` contains the root position.

**Reply by bcakmakoglu** [reply]
For nested nodes you'd have to overwrite the `computedPosition`
